.github/workflows/ci.yml

```diff
@@ -0,0 +1,87 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          # Test the latest stable release
+          - php-version: '7.1'
+            phpunit-version: 7
+          - php-version: '7.2'
+            phpunit-version: 8
+          - php-version: '7.3'
+            phpunit-version: 8
+          - php-version: '7.4'
+            phpunit-version: 8
+          - php-version: '8.0'
+            phpunit-version: 9
+          # Test Symfony LTS versions. Read more at https://github.com/symfony/lts
+          - php-version: '7.4'
+            dependencies: 'symfony/lts:^3'
+            phpunit-version: 8
+          - php-version: '8.0'
+            phpunit-version: 9
+            symfony-version: '^4'
+          - php-version: '8.0'
+            phpunit-version: 9
+            symfony-version: '^5'
+          - php-version: '8.0'
+            phpunit-version: 9
+            symfony-version: '^6'
+          # Minimum supported dependencies with the oldest PHP version
+          - php-version: '7.1'
+            phpunit-version: 7
+            composer-flag: '--prefer-stable --prefer-lowest'
+          # Test latest unreleased versions
+          - php-version: '8.0'
+            phpunit-version: 9
+            symfony-version: '^6'
+            stability: 'dev'
+    name: PHP ${{ matrix.php-version }} Test on Symfony ${{ matrix.symfony-version }} ${{ matrix.dependencies}} ${{ matrix.stability }} ${{ matrix.composer-flag }}
+    steps:
+      - name: Pull the code
+        uses: actions/checkout@v2
+      - name: Install PHP and Composer
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php-version }}
+          tools: composer:v2, phpunit:${{ matrix.phpunit-version }}
+          ini-values: phar.readonly=off
```

Comment:
this setting needs to be off, or we can't create a phar. it would make sense if the phar task checks the php.ini setting and outputs a more helpful error when readonly if on. but that is out of scope for this PR.

Reply:
Yep, it's a big out of scope. But maybe putting a comment above this line to explain why we need to have this could make sens

Reply:
adding a comment